src/builder/resettable.rs

```diff
@@ -83,6 +65,12 @@ impl<I: Into<StyledStr>> IntoResettable<StyledStr> for I {
     }
 }
 
+impl<I: Into<StyledStr>> IntoResettable<StyledStr> for Option<I> {
+    fn into_resettable(self) -> Resettable<StyledStr> {
+        self.map(Into::into).into()
+    }
+}
```

Comment:
> I noticed that currenlty only `Option<'static str>` implements `IntoResettable`.
> 
> The only reason I see would be that `Option::None` can infer its type that way. But this can be circumvented by using `Resettable::Reset` instead as I therefor updated all examples/tests/derivemacro to reflect that change.
> 
> The reason I implemented this change is to allow `Option<String>` as well.

Can you give the motivation for why you want `Option<String>` over using `Resettable<String>`?

Reply:
Because I'm getting them from a different place where they are already `Option<String>`.

But if the disadvantages of having this implementation outweigh the added flexibility, I could obviously just call `.map(Into::into)` myself.

Reply:
Support for runtime conditional values is secondary, so I'd like to see how things work out in the primary use case first.

I am open to more functions accepting `IntoResettable`.

Reply:
Makes sense, we can keep this open for now, but should decide before 4.0 as this would be a breaking change, due to the type inference.